main/src/org/jetbrains/bsp/bazel/resolvers/ProcessResolver.java

```diff
@@ -0,0 +1,79 @@
+package org.jetbrains.bsp.bazel.resolvers;
+
+import com.google.common.base.Splitter;
+import com.google.common.io.ByteStreams;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Semaphore;
+
+public class ProcessResolver {
+
+  private final Semaphore processLock;
```

Comment:
Just a minor note: this is a great solution for preventing any process concurrency issues but I was able to find that these weren't in any way a real issue, since all requests are done sequentially.

Reply:
There was a variable `processLock` which was used as ... a lock! (more precisely as a semaphore) I just changed it to the more elegant solution. Does it mean that it is redundant and even original `processLock` was redundant?

Reply:
As of that time yes it was, but since I hadn't tested this against other BSP implementations rather the IntelliJ one, I thought it wouldn't be a good idea to make assumptions

Reply:
In the general case we should assume that requests by a client aren't necessarily sequential, even if only for the sake of the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle)

Reply:
But since that isn't something we can take as guaranteed from clients (a lot of requests can be done in parallel), this is very much welcome, I like that you guys are doing so many changes I had in mind in terms of refactors :smile:

Reply:
Nice to hear this ☺️